src/DataWrangling/ECCO/ECCODarwin.jl

```diff
@@ -0,0 +1,485 @@
+    T     = eval(Symbol(meta_data.dataprec))
+    fid   = open(file_name)
+    ndims = meta_data.nDims
+    sizes = meta_data.dimList[:,1]
+    bin   = Array{T,1}(undef,*(meta_data.dimList[:,1]...));
+
+    read!(fid,bin)
+    bin   = hton.(bin)
+    close(fid)
+
+    ma    = read(
+                reshape(bin, sizes...),
+                mesh_grid,
+    )
+
+    #grid  = GridLoad(mesh_grid; option="full")
+    #lat=[j for i=-179.5:1.0:179.5, j=-89.5:1.0:89.5]
+    #lon=[i for i=-179.5:1.0:179.5, j=-89.5:1.0:89.5]
+    #(f,i,j,w)=InterpolationFactors(grid, vec(lon), vec(lat))
+    #coeffs=(lon=lon, lat=lat, f=f, i=i, j=j, w=w)
+    coeffs = interpolation_setup()
+    data   = Array{T,ndims}(
+                undef,
+                (size(coeffs.lon)...,size(ma,2)),
+    )
+
+    # Interpolate each layer
+    for z in 1:size(ma,2)
+        i, j, c = Interpolate(ma[:,z], coeffs)
+        data[:,:,z] = c
+    end
+
+    return data
+end
+
+"""
+    ECCO_field(metadata::ECCODarwinMetadata;
+               architecture = CPU(),
+               inpainting = nothing,
+               mask = nothing,
+               horizontal_halo = (7, 7),
+               cache_inpainted_data = false)
+
+Return a `Field` on `architecture` described by `ECCOMetadata` with
+`horizontal_halo` size.
+If not `nothing`, the `inpainting` method is used to fill the cells
+within the specified `mask`. `mask` is set to `ECCO_mask` for non-nothing
+`inpainting`.
+"""
+function ECCO_field(metadata::ECCODarwinMetadata;
+                    architecture = CPU(),
+                    inpainting = NearestNeighborInpainting(Inf),
+                    mask = nothing,
+                    horizontal_halo = (7, 7),
+                    cache_inpainted_data = true)
+
+    z = ClimaOcean.DataWrangling.ECCO.ECCO_z
+    input_data_grid = LatitudeLongitudeGrid(architecture; 
+                z,
+                size      = (720, 360, 50),
+                latitude  = (-90, 90),
+                longitude = (0, 360),
+    )
+    field_input_data = Field{Center, Center, Center}(input_data_grid)
+    inpainted_path = inpainted_metadata_path(metadata)
+
+    if !isnothing(inpainting) && isfile(inpainted_path)
+        file = jldopen(inpainted_path, "r")
+        maxiter = file["inpainting_maxiter"]
+
+        # read data if generated with the same inpainting
+        if maxiter == inpainting.maxiter
+            data = file["data"]
+            close(file)
+            copyto!(parent(field_input_data), data)
+            return field_input_data
+        end
+
+        close(file)
+    end
+
+    download_dataset(metadata)
+    path = metadata_path(metadata)
+
+    # Open ECCODarwin binary data file
+    data = ECCODarwinModelData(
+                path, 
+                ECCODarwinModelMeta(
+                    replace(path,"data"=>"meta"),
+                ),
+                ECCODarwinMeshGrid(
+                    metadata.version,
+                ),
+    )
```

Comment:
Here you can define a function `read_ECCO_data(metadata, path)` that dispatches on the version of the `metadata` and returns 
```julia
    shortname = short_name(metadata)

    if variable_is_three_dimensional(metadata)
        data = ds[shortname][:, :, :, 1]
        data = reverse(data, dims=3)
    else
        data = ds[shortname][:, :, 1]
    end        
```
by default and 
```julia
data = ECCODarwinModelData(
                path, 
                ECCODarwinModelMeta(
                    replace(path,"data"=>"meta"),
                ),
                ECCODarwinMeshGrid(
                    metadata.version,
                ),
    )
```
for `ECCODarwin*` versions. In this way you can reuse the current `ECCO_field` implementation

Reply:
Yes, that sounds like a plan.